modules/core/src/test/java/org/apache/ignite/internal/metric/CacheMetricsAddRemoveTest.java

```diff
@@ -82,17 +98,58 @@ public void testCacheMetricsAddRemove() throws Exception {
         checkMetricsEmpty(cachePrefix);
     }
 
+    /** */
+    @Test
+    public void testCacheMetricsNotRemovedOnStop() throws Exception {
+        String cachePrefix = cacheMetricsRegistryName("other-cache", false);
+
+        checkMetricsEmpty(cachePrefix);
+
+        createCache("persisted", "other-cache");
+
+        grid("client").cache("other-cache").put(1L, 1L);
+
+        checkMetricsNotEmpty(cachePrefix);
+
+        //Cache will be stopped during deactivation.
+        grid("client").cluster().state(ClusterState.INACTIVE);
+
+        checkMetricsNotEmpty(cachePrefix);
+
+        grid("client").cluster().state(ClusterState.ACTIVE);
+
+        assertEquals(1L, grid("client").cache("other-cache").get(1L));
+
+        checkMetricsNotEmpty(cachePrefix);
+
+        destroyCache();
```

Comment:
After cache destroying should be asserted that metric configuration is removed.

Reply:
Fixed.